File: app/src/main/java/org/yausername/dvd/utils/FileNameUtils.java
package org.yausername.dvd.utils;

public class FileNameUtils {

    public static String createFilename(String title) {
        String cleanFileName = title.replaceAll("[\\\\><\"|*?'%:#/]", " ");
        String fileName = cleanFileName.trim().replaceAll(" +", " ");
        if (fileName.length() > 127)
            fileName = fileName.substring(0, 127);

        return fileName;
    }

}

File: app/src/main/java/org/yausername/dvd/utils/FileNameUtils.kt
package org.yausername.dvd.utils

object FileNameUtils {
    fun createFilename(title: String): String {
        val cleanFileName = title.replace("[\\\\><\"|*?'%:#/]".toRegex(), " ")
        var fileName = cleanFileName.trim { it <= ' ' }.replace(" +".toRegex(), " ")
        if (fileName.length > 127) fileName = fileName.substring(0, 127)
        return fileName
    }
}